### PR TITLE
DYN-7475: Cherry picks for 2.16.4

### PR DESCRIPTION
**Pull request by twastvedt**
### Purpose

Cherry pick fixes agreed on for the 2.16.4 release. See [this wiki](https://wiki.autodesk.com/pages/viewpage.action?spaceKey=GEN&title=Dynamo+2.16.4) for more info. The fixes are:

DYN-5710 #13855 
DYN-6130 #14317 
DYN-6073 #14338 
DYN-5553, DYN-6103 #13820 
DYN-5810 #13908 
DYN-6158 #14314 
DYN-6123 #14220 
DYN-5314 #13647 
DYN-5640 #13773 

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. Use N/A to indicate that the changes in this pull request do not apply to Release Notes. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by QilongTang** [comment]
hmm even the EngOps build does not work and contains one build error, @twastvedt can you double check your change?

**Comment by twastvedt** [comment]
Yeah, sorry, bad cherry picks. Now builds, but some tests are failing. How do I run tests locally? Trying to just "Run Tests" from VS skips all the tests...

**Comment by QilongTang** [comment]
> Also I'm uncertain on how to fix the LaunchDarkly.Logging.dll file, reported [here](https://jira.autodesk.com/browse/DYN-7067). I _think_ we had fixed that in the original 2.16.3, which had build issues, so we ended up publishing a much smaller update, but now I'm not sure where to find the change that fixed that?

This might be a side effect switching from .NET8 back to .NET48..

**Comment by QilongTang** [comment]
I am giving the tests a try running at https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service_NUnit/16/

**Comment by twastvedt** [comment]
It builds! https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-Dynamo/job/DynamoBuildscripts/job/RC2.16.x_master/

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt [hunk 1]
```diff
@@ -69,7 +69,7 @@ using System.Runtime.InteropServices;
 <#+
 int MajorVersion = 2;
 int MinorVersion = 16;
-int BuildNumber = 3;
+int BuildNumber = 4;
```
Comment (QilongTang) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt:72]: 👍🏻

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 2]
```diff
@@ -533,6 +533,44 @@ private void RemoveGroupFromGroup(object parameters)
             Analytics.TrackEvent(Actions.RemovedFrom, Categories.GroupOperations, "Group");
         }
 
+        private void DissolveNestedGroups(object parameters)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:536]: Is this required? This is used by some commands not cherry-picked?

**Review by QilongTang — APPROVED** [review]
LGTM with one comment

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs (hunk 2 above)
Comment (twastvedt) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]: Thanks for catching that. Cherry-pick goofiness. I went back through all the commits and compared with the cherry-picked PRs again, looks like this is the only addition.

**Pull request merged**